### PR TITLE
React/deterministic

### PR DESCRIPTION
**Pull request by kzalite**
todo: refactor, cleanup old files

### CODE REVIEW

File: app/ts/DeterministicTab/DeterministicResults/SensitivityAnalysis/MeasurementSensitivity/MeasurementSensitivityPlot/MeasurementSensitivityPlot.tsx [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    measurementsSensitivityResults
+  } = useContext(DeterministicResultsContext);
+  const width = '400px';
+  const height = '400px';
+
+  const settings: ChartConfiguration = getSensitivityLineChartSettings(
+    measurementsSensitivityResults,
+    filteredAlternatives,
+    legendByAlternativeId,
+    measurementSensitivityCriterion.title,
```
Comment (kzalite) [inline app/ts/DeterministicTab/DeterministicResults/SensitivityAnalysis/MeasurementSensitivity/MeasurementSensitivityPlot/MeasurementSensitivityPlot.tsx:22]: refactor into plotSettings object?

File: app/ts/DeterministicTab/DeterministicResults/SensitivityMeasurementsTable/SensitivityMeasurementsTableRow/SensitivityMeasurementsTableCell/SensitivityMeasurementsTableCell.tsx [hunk 2]
```diff
@@ -0,0 +1,185 @@
...
+      return getPercentifiedValueLabel(values.currentValue, usePercentage);
+    }
+  }
+
+  return (
+    <TableCell id={`sensitivity-cell-${criterion.id}-${alternativeId}`}>
+      <Button className="text-centered" onClick={openPopover} variant="text">
+        <a> {getLabel()}</a>
+      </Button>
+      <Popover
```
Comment (kzalite) [inline app/ts/DeterministicTab/DeterministicResults/SensitivityMeasurementsTable/SensitivityMeasurementsTableRow/SensitivityMeasurementsTableCell/SensitivityMeasurementsTableCell.tsx:131]: Extract popover?

File: app/ts/DeterministicTab/DeterministicResults/SensitivityMeasurementsTable/SensitivityMeasurementsTableRow/SensitivityMeasurementsTableRow.tsx [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+
+export default function SensitivityMeasurementsTableRow({
+  criterion
+}: {
+  criterion: ICriterion;
+}): JSX.Element {
+  const {filteredAlternatives} = useContext(SubproblemContext);
+
+  function renderCells(): JSX.Element[] {
+    return _.map(filteredAlternatives, (alternative) => {
```
Comment (kzalite) [inline app/ts/DeterministicTab/DeterministicResults/SensitivityMeasurementsTable/SensitivityMeasurementsTableRow/SensitivityMeasurementsTableRow.tsx:21]: delete { return ...

File: app/ts/EffectsTable/EffectsTableUtil.ts [hunk 4]
```diff
@@ -9,3 +12,27 @@ export function canDSBePercentage(
     .unitOfMeasurement.type;
   return unitType === 'decimal' || unitType === 'percentage';
 }
+
+export function findValue<T extends Effect | Distribution>(
+  items: T[],
+  dataSourceId: string,
+  alternativeId: string
+): T {
+  return _.find(items, (item: T) => {
```
Comment (kzalite) [inline app/ts/EffectsTable/EffectsTableUtil.ts:21]: delete return { ...

File: test/endToEnd/scenarioTest.js [hunk 5]
```diff
@@ -112,7 +112,9 @@ function switchInDeterministic(browser) {
     .click('#deterministic-tab')
     .waitForElementVisible('#sensitivity-measurements-header')
     .click('#scenario-selector')
-    .click('option[label="Default"]')
+    .waitForElementVisible('#scenario-selector > option:nth-child(1)')
```
Comment (DanielReid) [inline test/endToEnd/scenarioTest.js:115]: nth-child

**Review by DanielReid — APPROVED** [review]
seems good

File: test/endToEnd/scenarioTest.js (hunk 5 above)
Comment (kzalite) [reply test/endToEnd/scenarioTest.js:115]: label is scenario id. can put value on the component, but...

File: test/endToEnd/closeModalTest.js [hunk 6]
```diff
@@ -230,10 +230,10 @@ function cancelEditingGraphLabels(browser) {
   var paths = {
     valueToClear: '#label-input-0',
     tab: '#deterministic-tab',
-    actionButton: '//*[@id="value-profile-container"]/div[2]/button',
+    actionButton: '//*[@id="base-profile-plot"]/div[2]/button',
```
Comment (chinook25) [inline test/endToEnd/closeModalTest.js:233]: can't we just put an id on the button itself?

File: test/endToEnd/deterministicResultsTest.js [hunk 7]
```diff
@@ -57,55 +35,62 @@ function afterEach(browser) {
...
+    .waitForElementVisible('#preferences-sensitivity-plot');
+
+  const measurementValuePath =
+    '//*[@id="sensitivity-cell-treatmentRespondersId-placeboId"]/button/span[1]';
+  const weightValuePath =
+    '//*[@id="deterministic-weights-table"]/tbody/tr/td[1]';
+  const baseTotalValuePath =
+    '//*[@id="base-total-value-table"]/table/tbody/tr/td[1]';
+  const baseValueProfilePath =
+    '//*[@id="base-value-profiles-table"]/table/tbody/tr[1]/td[2]';
```
Comment (chinook25) [inline test/endToEnd/deterministicResultsTest.js:52]: ah so there are tests for it :P

File: app/ts/Legend/LegendWrapper/LegendWrapper.tsx [hunk 8]
```diff
@@ -1,21 +1,23 @@
...
 export default function LegendWrapper({
-  children
+  children,
+  buttonId
 }: {
   children: any;
+  buttonId:string
 }): JSX.Element {
   const {canEdit, legendByAlternativeId, saveLegend} = useContext(
     LegendContext
```
Comment (chinook25) [inline app/ts/Legend/LegendWrapper/LegendWrapper.tsx:23]: needs autolayout

File: app/ts/Legend/LegendWrapper/LegendWrapper.tsx [hunk 9]
```diff
@@ -79,6 +81,7 @@ export default function LegendWrapper({
       <Grid container item md={12} lg={4} alignContent="flex-start">
         <Tooltip title={<LegendTooltip />}>
           <Button
+          id={buttonId}
```
Comment (chinook25) [inline app/ts/Legend/LegendWrapper/LegendWrapper.tsx:84]: layout

**Pull request merged**